### ISSUE TITLE
Make utilities for registering "special types" public

### ISSUE THREAD

**Issue by mskoenz** [FrescolinoGroup/pyhdf5io#11]
(edited by @greschd)

Several helpers used for registering "special types" are not public, but should be:
- ``add_type_tag``
- ``_value_serializer``
- ``_ValueDeserializer``
and maybe others.

These might need renaming, improving the docstring, and some documentation.

**Comment by mskoenz** [comment]
Does not need to be fixed right now, just wanted to note it

**Comment by mskoenz** [comment]
its not only about special types, it about hdf5_io support without deriving
(and hence being able to use the entry_points)

**Comment by greschd** [comment]
Yeah, that's what the "special types" are about. If there is a `to_hdf5` / `from_hdf5` method in the class, that _always_ takes precedence over the singledispatch mapping.

Special, as in, doesn't have built-in hdf5-io support.
